libs/shinkai-node-state/src/v2/mutations/duplicateTool/useDuplicateTool.ts

```diff
@@ -0,0 +1,17 @@
+import { useMutation } from "@tanstack/react-query";
+
+import { duplicateTool } from ".";
+
+export const useDuplicateTool = ({
+  toolKey,
+  nodeAddress,
+  token,
+}: {
+  toolKey: string;
+  nodeAddress: string;
+  token: string;
+}) => {
+  return useMutation({
+    mutationFn: () => duplicateTool({ toolKey, nodeAddress, token }),
+  });
+};
```

Comment:
useMutation have diff arguments/params from useQuery (eg: onSuccess, onError, etc) which doesnt exist in queries. Ideally what we want is to have access to these callbacks to show either a toast or invalidating queries, you can look into `useEnableAllTools` for some example and how its being used in `tool-collection.tsx`

Reply:
Done, I guess the key to invalidate is also GET_LIST_TOOLS since its adding a new tool to the list